integration-cli/daemon/daemon.go

```diff
@@ -757,6 +759,16 @@ func (d *Daemon) ReloadConfig() error {
 	return nil
 }
 
+// NewClient creates new client based on daemon's socket path
+func (d *Daemon) NewClient() (*client.Client, error) {
+	var httpClient *http.Client
```

Comment:
@dnephin just so I understand what is the reason for this? if we specify to use d.Sock() in client.NewClient anyway(line 765). I tried it and it works but I don't see a difference